### PR TITLE
drt: functions changed to have more explicit output

### PR DESCRIPTION
**Pull request by bnmfw**
This PR contain minor changes to some functions in PA. Functions changed take a reference to a container and fill said container. This PR changes it so instead it returns it.

This makes the functions role easier to understand, as its more obvious that its role is to return that structure and that it is an output, not an input.

This should not affect performance as due to **move semantics** instead of copying the returned structure its ownership is transferred to the caller.

Some functions have name changes:
- `prepPoint_pin_mergePinShapes` → `mergePinShapes` (Conform to PR [#5770](https://github.com/The-OpenROAD-Project/OpenROAD/pull/5770))
- `initMaster2PinLayerRange` → `initMasterToPinLayerRange`

Also `LayerRange` was changed to a pair instead of a tuple, as its always a tuple of 2 elements. This does not affect its use in code as it is accessed using `auto [a, b] = LayerRange`.

### CODE REVIEW

File: src/drt/src/pa/FlexPA_init.cpp [hunk 1]
```diff
@@ -48,14 +48,13 @@ void FlexPA::initViaRawPriority()
     for (auto& via_def :
          design_->getTech()->getLayer(layer_num)->getViaDefs()) {
       const int cutNum = int(via_def->getCutFigs().size());
-      ViaRawPriorityTuple priority;
-      getViaRawPriority(via_def, priority);
+      ViaRawPriorityTuple priority = getViaRawPriority(via_def);
```
Comment (maliberty) [inline src/drt/src/pa/FlexPA_init.cpp:51]: can be const now

File: src/drt/src/pa/FlexPA_prep.cpp [hunk 2]
```diff
@@ -113,11 +111,11 @@ void FlexPA::prepPoint_pin_mergePinShapes(
       using boost::polygon::operators::operator+=;
       pin_shapes[layer_num] += poly;
     } else {
-      logger_->error(
-          DRT, 67, "FlexPA prepPoint_pin_mergePinShapes unsupported shape.");
+      logger_->error(DRT, 67, "FlexPA mergePinShapes unsupported shape.");
       exit(1);
```
Comment (maliberty) [inline src/drt/src/pa/FlexPA_prep.cpp:115]: Please remove the `exit(1)` it is deadcode (and we should never call exit).

**Review by maliberty — DISMISSED** [review]

**Comment by maliberty** [comment]
I'm satisfied but want to leave a chance for @osamahammad21 to comment before merging.

File: src/drt/src/pa/FlexPA_unique.h [hunk 3]
```diff
@@ -69,21 +69,21 @@ class UniqueInsts
   void setDesign(frDesign* design) { design_ = design; }
 
  private:
-  using LayerRange = std::tuple<frLayerNum, frLayerNum>;
+  using LayerRange = std::pair<frLayerNum, frLayerNum>;
   using MasterLayerRange = std::map<frMaster*, LayerRange, frBlockObjectComp>;
```
Comment (osamahammad21) [inline src/drt/src/pa/FlexPA_unique.h:73]: Use frBlockObjectMap instead, not necessary in this PR.

**Review by osamahammad21 — APPROVED** [review]

**Review by osamahammad21 — APPROVED** [review]

**Pull request merged**